tests/pipeline/test_pipeline.py

```diff
@@ -240,12 +240,51 @@ def test_with_custom_config(self, tmpdir, snapshot):
 
         enriched_pipeline = load_yaml_file(Path(output_file_path))
         assert isinstance(enriched_pipeline, dict)
-        for app_details in enriched_pipeline["components"]:
-            output_topic = app_details["app"]["streams"]["outputTopic"]
-            assert output_topic == "random-topic"
-            app_type = app_details["type"]
-            if app_type == "streams-app":
-                error_topic = app_details["app"]["streams"]["errorTopic"]
-                assert error_topic == "random-error"
+
+        producer_details = enriched_pipeline["components"][0]
+        output_topic = producer_details["app"]["streams"]["outputTopic"]
+        assert output_topic == "random-topic"
+
+        streams_app_details = enriched_pipeline["components"][1]
+        output_topic = streams_app_details["app"]["streams"]["outputTopic"]
+        assert output_topic == "random-topic"
+        error_topic = streams_app_details["app"]["streams"]["errorTopic"]
+        assert error_topic == "random-error"
+
+        snapshot.assert_match(enriched_pipeline, "test-pipeline")
+
+    def test_default_config(self, tmpdir, snapshot):
+        os.environ["KPOPS_ENVIRONMENT"] = "development"
+        output_file_path = tmpdir.join("pipeline.yaml")
+        result = runner.invoke(
+            app,
+            [
+                "generate",
+                "--pipeline-base-dir",
+                PIPELINE_BASE_DIR,
+                str(RESOURCE_PATH / "no-topics/pipeline.yaml"),
+                "--save",
+                "--out-path",
+                output_file_path,
+                "--defaults",
+                str(RESOURCE_PATH / "no-topics-defaults"),
+                "--config",
+                str(RESOURCE_PATH / "no-topics/config.yaml"),
+            ],
+        )
+        assert result.exit_code == 0, result.exception
```

Comment:
```suggestion
        assert result.exit_code == 0
```

this is no longer needed if the runner doesn't catch exceptions